neurobooth_os/iout/flir_cam.py

```diff
@@ -190,44 +189,45 @@ def prepare(self, name="temp_video"):
         with meta.get_database_connection() as db_conn:
             meta.post_message(Request(source='Flir', destination='CTR', body=msg_body), conn=db_conn)
         self.streaming = True
+        self.manifest_dict["frame_rate"] = self.FRAME_RATE_OUT
+        self.manifest_dict['frame_height'] = im.shape[0]
+        self.manifest_dict['frame_width'] = im.shape[1]
+        self.manifest_dict['frame_depth'] = im.shape[2]
 
     def record(self):
         self.logger.debug('FLIR: LSL Thread Started')
         self.recording = True
-        self.frame_counter = 0
-        self.save_thread = threading.Thread(target=self.camCaptureVid)
-        self.save_thread.start()
-
         self.stamp = []
-        while self.recording:
-            # Exception for failed waiting self.cam.GetNextImage(1000)
-            try:
-                im, tsmp = self.imgage_proc()
-            except:
-                continue
-
-            self.image_queue.put(im)
-            self.stamp.append(tsmp)
-
-            try:
-                self.outlet.push_sample([self.frame_counter, tsmp])
-            except BaseException:
-                self.logger.debug(f"Reopening FLIR {self.device_index} stream already closed")
-                self.outlet = self.createOutlet(self.video_filename)
-                self.outlet.push_sample([self.frame_counter, tsmp])
-
-            # self.video_out.write(im_conv_d)
-            self.frame_counter += 1
+        self.frame_counter = 0
 
-            if not self.frame_counter % 1000 and self.image_queue.qsize() > 2:
-                self.logger.debug(
-                    f"Queue length is {self.image_queue.qsize()} frame count: {self.frame_counter}"
-                )
+        # write flir manifest
+        manifest_file_name = self.video_filename.replace(".images", "_manifest.yaml")
+        self.manifest_dict["image_file"] = self.video_filename
+        with open(manifest_file_name, "w") as file:
+            yaml.dump(self.manifest_dict, file)
+
+        with open(self.video_filename, 'wb', buffering=10485760) as file:
```

Comment:
I'd expose the buffering constant to the top of the script and add a comment to indicate what its for or what it means, basically add a bit more context. Is this a fixed constant or something tunable that needs to live in the device configs?

Reply:
will do